RUFAS/routines/animal/life_cycle/heiferII.py

```diff
@@ -310,390 +337,983 @@ def update(self, sim_day):
-            else:
-                self.events.add_event(self.days_born, sim_day, const.HEIFER_NOT_PREG)
-        # preg check 1
-        elif self.days_born == self.ai_day + AnimalBase.config["preg_check_day_1"]:
-            self.preg_diagnoses += 1
-
-            if self.days_in_preg > 0:
-                preg_loss_rand = random()
-                if preg_loss_rand > AnimalBase.config["preg_loss_rate_1"]:
-                    self.events.add_event(self.days_born, sim_day, const.PREG_CHECK_1_PREG)
-                else:
-                    self.days_in_preg = 0
-                    self.abortion_day = self.days_born
-                    self.open(sim_day)
-                    self.body_weight -= self.conceptus_weight
-                    self.conceptus_weight = 0
-                    self.calf_birth_weight = 0
-                    self.p_gest_for_calf = 0
-                    self.events.add_event(self.days_born, sim_day, const.PREG_LOSS_BEFORE_1)
-            else:
-                self.abortion_day = self.days_born
-                self.open(sim_day)
-                self.events.add_event(self.days_born, sim_day, const.PREG_CHECK_1_NOT_PREG)
-        # preg check 2
-        elif self.days_born == self.ai_day + AnimalBase.config["preg_check_day_2"]:
-            self.preg_diagnoses += 1
-            preg_loss_rand = random()
-            if preg_loss_rand > AnimalBase.config["preg_loss_rate_2"]:
-                self.events.add_event(self.days_born, sim_day, const.PREG_CHECK_2_PREG)
-            else:
-                self.days_in_preg = 0
-                self.abortion_day = self.days_born
-                self.open(sim_day)
-                self.body_weight -= self.conceptus_weight
-                self.conceptus_weight = 0
-                self.calf_birth_weight = 0
-                self.p_gest_for_calf = 0
-                self.events.add_event(self.days_born, sim_day, const.PREG_LOSS_BTWN_1_AND_2)
-        # preg check 3
-        elif self.days_born == self.ai_day + AnimalBase.config["preg_check_day_3"]:
-            self.preg_diagnoses += 1
-            preg_loss_rand = random()
-            if preg_loss_rand > AnimalBase.config["preg_loss_rate_3"]:
-                self.events.add_event(self.days_born, sim_day, const.PREG_CHECK_3_PREG)
+        return AnimalBase.config['heifer_repro_method']
+
+    @staticmethod
+    def get_user_defined_repro_sub_protocol() -> str:
+        """
+        Get the reproduction sub protocol for heifers defined by the user.
+
+        Returns
+        -------
+        str
+            The reproduction sub protocol for heifers defined by the user.
+        """
+
+        return HeiferII.get_user_defined_repro_data('repro_sub_protocol')
+
+    @staticmethod
+    def _get_default_repro_sub_protocol(protocol: str) -> str:
+        """
+        Get the default reproduction sub protocol for heifers for a given reproduction protocol.
+
+        This is defined in the InternalReproSettings class.
+
```

Comment:
This should be changed to either
```suggestion
        Get the default reproduction sub protocol for heifers for a given reproduction protocol, defined in the InternalReproSettings class.

```
or 
```suggestion
        Get the default reproduction sub protocol for heifers for a given reproduction protocol.

        Notes
        -----
        The default reproduction sub protocol for heifers is defined in the InternalReproSettings class.

```